packages/react-router-dom/index.tsx

```diff
@@ -430,8 +432,8 @@ export const Link = React.forwardRef<HTMLAnchorElement, LinkProps>(
       // eslint-disable-next-line jsx-a11y/anchor-has-content
       <a
         {...rest}
-        href={href}
-        onClick={reloadDocument ? onClick : handleClick}
+        href={isExternal ? toString : href}
+        onClick={isExternal || reloadDocument ? onClick : handleClick}
```

Comment:
Check out this semi-related bug that got reported over the weekend: https://github.com/remix-run/react-router/pull/9913.  Remix handled same-origin absolute URLs as internal client-side transitions (`redirect('https://www.currentorigin.com/path')`) which we missed in 1.10.  

We should probably keep the same behavior on `<Link to>` since they can always opt-into hard-reload with `reloadDocument`?  Should just need to strip off the origin in `toString` before `useHref` if it matches current origin (I think 😉 )